### PR TITLE
Se 147

### PR DESCRIPTION
**Pull request by PeterAlabaster**
[print: Extend automated printing hook]

When an automated print action occurrs, log a note in the audit
trail of the records which were printed. Only do this if
the class attribute AUDIT_LOG_PRINTING exists and is set to True

[print: Extend web load controller]

When a desktop print action occurrs, log a note in the audit
trail of the records which were printed. Only do this if
the class attribute AUDIT_LOG_PRINTING exists and is set to True

### CODE REVIEW

**Comment by PeterAlabaster** [comment]
image

**Review by rajeeshknambiar — APPROVED** [review]

File: addons/print/models/ir_actions_print.py [hunk 1]
```diff
@@ -64,6 +64,17 @@ def _run_action_print(self, eval_context=None):  # pylint: disable=unused-argume
                     active_id,
                 )
                 printer.spool_report(records.ids, report)
+                # Log a note in the audit trail
+                # of the affected records when a report is direct printed.
+                # Only affects models who have the class attribute AUDIT_LOG_PRINTING set to True.
+
+                # NB: If setting this class attribute on models,
+                # ensure the model has `_inherit = ["mail.thread"]`
+                if hasattr(obj, "AUDIT_LOG_PRINTING") and obj.AUDIT_LOG_PRINTING == True:
```
Comment (kdwyer) [inline addons/print/models/ir_actions_print.py]: Nit: could this just be
```python
hasattr(obj, "AUDIT_LOG_PRINTING") and obj.AUDIT_LOG_PRINTING
```
or if it must be a bool
```python
hasattr(obj, "AUDIT_LOG_PRINTING") and obj.AUDIT_LOG_PRINTING is True
```
?

Comment (kdwyer) [reply addons/print/models/ir_actions_print.py]: (same would apply to the other commit)

Comment (PeterAlabaster) [reply addons/print/models/ir_actions_print.py]: Changed :)

**Review by kdwyer — APPROVED** [review]

**Pull request merged**